java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertVarToExplicitType.java

```diff
@@ -152,16 +153,18 @@ private static boolean isLocalVarType(HintContext ctx) {
         }
 
         TreePath treePath = ctx.getPath();
-        EnhancedForLoopTree efl = null;
-        TypeMirror expressionType = null;
-        if (treePath.getLeaf() instanceof EnhancedForLoopTree) {
-            efl = (EnhancedForLoopTree) treePath.getLeaf();
-            expressionType = ctx.getInfo().getTrees().getTypeMirror(new TreePath(treePath, efl.getExpression()));
-        }
-        // should be local variable
-        // Can either be a local variable or iterator in enhanced-for-loop
-        if ((info.getTrees().getElement(treePath) != null && info.getTrees().getElement(treePath).getKind() != ElementKind.LOCAL_VARIABLE) && expressionType == null) {
-            return false;
+        if (treePath.getLeaf().getKind() == Tree.Kind.ENHANCED_FOR_LOOP) {
```

Comment:
Lines 156-167 seem identical to `ConvertToVarHint.java:175-187`. The classes are in the same package; could the code be shared (I think no instance vars are used).

Reply:
Yes, this code can be shared. I have made this change in the latest commit.